Bottom: 887c203d037bec6ef0fa36e8f4e399228302461e
Top:    9078c53a5a07b252d5a11d88a930d96cf6a7aadb
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 00:53:25 +0000

Enable output of via pads, and dial down circular approximation accuracy


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 92a3a1e..1b958d98 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1210,6 +1210,31 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
         }
 #endif
 
+      /* TODO: Inter-layer features
+       *
+       * Accumulate a circular polygon for each plated hole we may cut, ensuring
+       * the finished polygon contour extends to include the via barrel extents.
+       *
+       * Subtract non-plated hole contours from the polygons.
+       *
+       * For each hole, add the via-barrel between layers... removing the Object3D
+       * from the list of objects as they become joined with some other. (The final
+       * list of objects shuold match 1:1 with resultant bodies, and contain no
+       * duplicates.
+       *
+       * To accomodate overlapping drill holes, accumulate all via-barrels into a
+       * polygon, and subtract that from the positive copper polygon. As we already
+       * added via-barrels to each copper layer, the each barrel extrusion contour
+       * should match only one body of copper on a given layer.
+       *
+       * Remove the drilled hole down each via by extruding the additional faces.
+       *
+       * To accomodate overlapping drill holes, accumulate all drills into a polygon,
+       * and subtract that from the positive copper polygon. Any subtracted contour
+       * should at this point match to one body of copper on a given layer.
+       *
+       */
+
       if (info.poly == NULL)
         {
           fprintf (stderr, "Skipping layer group %i, info.poly was NULL\n", group);
